src/main/java/com/google/sps/data/ComputeAPIClient.java

```diff
@@ -0,0 +1,304 @@
+    } catch (IOException e) {
+        System.out.println("Got error!");
+        System.err.println(e.getMessage());
+    } catch (Throwable t) {
+        t.printStackTrace();
+      }
+  }
+
+  public void createInstance(String instanceName, String vncServerPassword)
+    throws IOException {
+      Instance instance = new Instance();
+      instance.setName(instanceName);
+      instance.setMachineType(String.format(
+        "projects/%s/zones/%s/machineTypes/n1-standard-1",
+        PROJECT_ID, ZONE_NAME));
+      setUpNetworkInterace(instance);
+      setUpDisk(instance);
+      setUpMetadata(instance, vncServerPassword);
+      setUpServiceAccounts(instance);
+      System.out.println(instance.toPrettyString());
+      Compute.Instances.Insert create = 
+        compute.instances().insert(PROJECT_ID, ZONE_NAME, instance);
+      Operation op = create.execute();
+      waitForOperationCompletion(op);
+    } 
+
+  public void stopInstance(String instanceName) throws Exception {
+    Compute.Instances.Stop stop =
+      compute.instances().stop(PROJECT_ID, ZONE_NAME, instanceName);
+    Operation op = stop.execute();
+    waitForOperationCompletion(op);
+  }
+
+  public void restartInstance(String instanceName, String vncServerPassword)
+    throws Exception {
+      Compute.Instances.Get getInstance = 
+        compute.instances().get(PROJECT_ID, ZONE_NAME, instanceName);
+      Instance instance = getInstance.execute();
+      String getFingerPrints = instance.getMetadata().getFingerprint();
+      Metadata metadata = createMetadata(vncServerPassword);
+      metadata.setFingerprint(getFingerPrints);
+      Compute.Instances.SetMetadata setMeta = 
+        compute.instances().setMetadata
+        (PROJECT_ID, ZONE_NAME, instanceName, metadata);
+      setMeta.execute();
+      Compute.Instances.Start restart =
+        compute.instances().start(PROJECT_ID, ZONE_NAME, instanceName);
+      Operation op = restart.execute();
+      waitForOperationCompletion(op);
+    }
+
+  public void deleteInstance(String instanceName) throws Exception {
+    Compute.Instances.Delete delete =
+      compute.instances().delete(PROJECT_ID, ZONE_NAME, instanceName);
+    Operation op = delete.execute();
+    waitForOperationCompletion(op);
+  }
+
+  /**
+   * Waits until {@code operation} is completed.
+   * @param {Compute} compute - the {@code Compute} object
+   * @param {Operation} operation - the operation returned by the 
+   *    original request
+   * @param {long} timeout - the timeout, in millis
+   * @return the error, if any, else {@code null} if there was no error
+   * @throws InterruptedException if we timed out waiting for the operation
+   *    to complete
+   * @throws IOException if we had trouble connecting
+   */
+  private Operation.Error blockUntilComplete(
+    Compute compute, Operation operation, long timeout) throws Exception {
+      long start = System.currentTimeMillis();
+      final long pollIntervalMs = 5 * 1000;
```

Comment:
Also make this a constant. 
private static final int POLL_INTERVAL_MS = 5 * 1000;  // 5 seconds